src/app/users/users.component.ts

```diff
@@ -120,22 +118,30 @@ export class UsersComponent implements OnInit, AfterViewInit {
   }
 
   addRole(user) {
-    // If user has no previous role, add learner role
-    let selectedRolesArray = this.selectedRolesMap.get(user.name);
-    if (this.selectedRolesMap.get(user.name) === undefined || this.selectedRolesMap.get(user.name).length === 0) {
-      selectedRolesArray = [ 'learner' ];
-    }
-    const tempUser = { ...user, roles: [ ...selectedRolesArray ] };
-    this.couchService.put('_users/org.couchdb.user:' + tempUser.name, tempUser).subscribe((response) => {
-      console.log('Success!');
-      this.initializeData();
+    let oldRoles: string[] = [];
+    let selectedRole: string[] = [];
+    this.couchService.get('_users/org.couchdb.user:' + user.name).subscribe((data) => {
+      oldRoles = data.oldRoles;
+      if ( oldRoles.length === 0 ) {
+        selectedRole = [ 'learner' ];
+      } else {
+        selectedRole = oldRoles;
+      }
+      const tempUser = { ...user, roles: [ ...selectedRole ] };
+      this.couchService.put('_users/org.couchdb.user:' + tempUser.name, tempUser).subscribe((response) => {

```

Comment:
Based on my other feedback, there should only be one couchService request, but for future reference:

We should not nest observable subscribes.  We can use the rxjs operator `switchMap` ([documentation](http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#instance-method-switchMap)) to chain observables in a sequence if needed.